assets/js/base/context/hooks/use-checkout-address.ts

```diff
@@ -101,8 +82,8 @@ export const useCheckoutAddress = (): CheckoutAddress => {
 			! forcedBillingAddress && needsShipping && ! prefersCollection,
 		showShippingMethods: needsShipping && ! prefersCollection,
 		showBillingFields:
-			! needsShipping || ! useShippingAsBilling || prefersCollection,
+			! needsShipping || ! useShippingAsBilling || !! prefersCollection,
```

Comment:
TS fix. `prefersCollection` can be either a boolean or undefined, this accounts for and assumes `undefined` is `false` so that `showBillingFields` and `useBillingAsShipping` are always boolean.